src/WebJobs.Extensions.DurableTask/Scale/DurableTaskTriggersScaleProvider.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#nullable enable
+#if FUNCTIONS_V3_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Scale
+{
+    internal class DurableTaskTriggersScaleProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly IScaleMonitor monitor;
+        private readonly ITargetScaler targetScaler;
+
+        public DurableTaskTriggersScaleProvider(
+            IServiceProvider serviceProvider,
```

Comment:
nit: I think it's considered good practice to declare your dependent services explicitly in your constructor rather than taking `IServiceProvider` and fetching them manually. It certainly makes the code more unit testable.

For example, you could replace `IServiceProvider` here with parameters for `INameResolver`, `ILoggerFactory`, `IOptions<DurableTaskOptions>`, and `IEnumerable<IDurabilityProviderFactory>`.